Sources/KarrotListKit/Adapter/CollectionViewAdapter.swift

```diff
@@ -233,30 +231,94 @@ final public class CollectionViewAdapter: NSObject {
+    let layout = collectionView?.collectionViewLayout as? UICollectionViewCompositionalLayout
+    return layout?.configuration.scrollDirection ?? .vertical
+  }
+
+  /// Checks if the collection view has reached the end, and triggers the `ReachedEndEvent` if needed.
+  ///
+  /// This method manually evaluates if the collection view is near the end, based on the current content offset and view bounds.\
+  /// Should call this method on `scrollViewDidScroll(_:)` function of `UIScrollViewDelegate`.\
+  /// Basically, the `ReachedEndEvent` check is handled in the `scrollViewWillEndDragging` method.
+  private func manuallyCheckReachedBottomEventIfNeeded() {
+    guard
+      let collectionView,
+      collectionView.isDragging == false,
+      collectionView.isTracking == false
     else {
       return
     }
+    triggerReachedBottomEventIfNeeded(contentOffset: collectionView.contentOffset)
+  }
 
-    guard trigger.threshold >= section.cells.count - indexPath.item else {
+  /// Evaluates the position of the content offset and triggers the `ReachedEndEvent` if the end of the content is near.
+  ///
+  /// - Parameter contentOffset: The current offset of the content view.
+  ///
+  /// This method calculates the distance to the end of the content, considering the current scroll direction (vertical or horizontal).\
+  /// It computes the view length, content length, and offset based on the scroll direction. If the content length is smaller than the view length,\
+  /// it immediately triggers the `ReachedEndEvent`. Otherwise, it calculates the remaining distance and compares it to the trigger distance.\
+  /// If the remaining distance is less than or equal to the trigger distance, the `ReachedEndEvent` is triggered.\
+  private func triggerReachedBottomEventIfNeeded(contentOffset: CGPoint) {
+    guard
+      let event = list?.event(for: ReachedEndEvent.self),
+      let collectionView, collectionView.bounds.isEmpty == false
+    else {
       return
     }
 
-    trigger.context.state = .triggered
-    trigger.handler(trigger.context)
-  }
+    let viewLength: CGFloat
+    let contentLength: CGFloat
+    let offset: CGFloat
 
-  // MARK: - Action Methods
+    switch scrollDirection {
+    case .vertical:
+      viewLength = collectionView.bounds.size.height
+      contentLength = collectionView.contentSize.height
+      offset = contentOffset.y
 
-  @objc
-  private func pullToRefresh() {
-    list?.event(for: PullToRefreshEvent.self)?.handler(.init())
+    default:
+      viewLength = collectionView.bounds.size.width
+      contentLength = collectionView.contentSize.width
+      offset = contentOffset.x
+    }
+
+    if contentLength < viewLength {
+      event.handler(.init())
+      return
+    }
+
+    let triggerDistance: CGFloat = {
+      switch event.offset {
+      case .absolute(let offset):
+        return offset
+      case .relativeToContainerSize(let relativeToContainerSize):
+        return viewLength * relativeToContainerSize
```

Comment:
p3; 연관값 이름 그대로 바인딩하면 좋을 것 같네요!

```suggestion
      case .relativeToContainerSize(let multiplier):
        return viewLength * multiplier
```

Reply:
앗 replace하다가 한꺼번에 변경이 되었군여!! 좋습니다!

Reply:
amend push 해서 반영했어요~

[c51243c](https://github.com/daangn/KarrotListKit/pull/44/commits/c51243c950fa1df1960c3d1907eb677e92b396dc)에서 확인 부탁드려용!